Migrate list and show page to GOVUK Design systems

DilwoarH opened the pull request:
## What
Migrate the List and show page to the GOVUK design system

## Why
Migrating to GOVUK Design System will make it easier for users to use, consistent with the rest of GOVUK, and more accessible

## Before

<img width="1268" alt="Screenshot 2022-04-06 at 3 15 53 am" src="https://user-images.githubusercontent.com/4599889/161882440-2de1fd9b-2faa-4d1c-8543-0969cf9863f0.png">
<img width="1232" alt="Screenshot 2022-04-06 at 3 15 40 am" src="https://user-images.githubusercontent.com/4599889/161882453-14661ee5-c868-4585-a874-4c776b744ad2.png">

## After

<img width="1104" alt="Screenshot 2022-04-06 at 3 15 10 am" src="https://user-images.githubusercontent.com/4599889/161882488-7a85a02c-3b20-43ad-a393-427b8e7c5203.png">
<img width="1079" alt="Screenshot 2022-04-06 at 3 15 24 am" src="https://user-images.githubusercontent.com/4599889/161882502-a1e4e088-2fb5-460e-bccf-5a512d23afc9.png">


https://trello.com/c/sFromAP6/352-travel-advice-list-index-page

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

davidgisbey commented on app/views/admin/countries/_show-legacy.html.erb:
```diff
@@ -0,0 +1,40 @@
+<ul class="breadcrumb">
```
Normally partials use underscores rather than dashes in the file name

DilwoarH replied:
done

davidgisbey commented on app/views/admin/countries/index.html.erb:
```diff
@@ -1,34 +1,23 @@
-<h1 class="page-title">
-  Countries
-</h1>
+<%= render :partial => 'index-legacy' if @legacy_layout %>
```
Not a massive deal, but I think it's a bit cleaner to handle this in the controller. I'll leave a comment there

DilwoarH replied:
good idea - done

davidgisbey commented on app/controllers/admin/countries_controller.rb:
```diff
@@ -1,16 +1,29 @@
 
-  def show; end
+  def show
+    @legacy_layout = is_legacy_layout?
```
render `show_legacy` if is_legacy_layout?

Then change move the partial into a new view

DilwoarH replied:
good idea

DilwoarH replied:
done

davidgisbey commented on app/controllers/admin/countries_controller.rb:
```diff
@@ -1,16 +1,29 @@
 
   def index
     @countries = Country.all
+    @legacy_layout = is_legacy_layout?
```
render `show_index` if is_legacy_layout?

Then change move the partial into a new view. That way the views are decoupled.

Wdyt?

DilwoarH replied:
done

davidgisbey commented on app/controllers/admin/countries_controller.rb:
```diff
@@ -1,16 +1,29 @@
 private
 
+  def is_legacy_layout?
+    ! preview_design_system_user?
```
`!preview_design_system_user?`

in ruby

davidgisbey reviewed (approved):
lgtm

davidgisbey commented:
Yea all good. I've gone for a slightly different approach https://github.com/alphagov/travel-advice-publisher/pull/1302/commits/f84f92c0f43d0812e55aa6e530ec19cde9845f76

where i added some helper methods and split the specs out into different files (mainly because the specs are already massive so it'll be easier to delete them). I don't think it's overly necessary with yours as there are so few tests but up to you.

DilwoarH commented:
> Yea all good. I've gone for a slightly different approach [f84f92c](https://github.com/alphagov/travel-advice-publisher/commit/f84f92c0f43d0812e55aa6e530ec19cde9845f76)
> 
> where i added some helper methods and split the specs out into different files (mainly because the specs are already massive so it'll be easier to delete them). I don't think it's overly necessary with yours as there are so few tests but up to you.

Which approach do you think we should follow?

I followed the approach used in content publisher: https://github.com/alphagov/content-publisher/blob/c8d93843e99318d4180ce93ed4e07abb03395121/spec/requests/editions_spec.rb#L29-L31

davidgisbey commented:
Generally if you're reusing a bit of code repeatedly throughout multiple spec files you add a helper method. If it's just once or twice then adding it like Content Publisher is fine. Having said that it's somewhat nit picky and not a blocking change.
